R/compiled.R

```diff
@@ -72,27 +72,56 @@ run_gcov <- function(path, quiet = TRUE, clean = TRUE,
      return()
   }
 
-  gcov_inputs <- list.files(path, pattern = rex::rex(".gcno", end), recursive = TRUE, full.names = TRUE)
+  withr::local_dir(src_path)
+
+  gcov_inputs <- list.files(".", pattern = rex::rex(".gcno", end), recursive = TRUE, full.names = TRUE)
+
   if (!nzchar(gcov_path)) {
     if (length(gcov_inputs)) stop('gcov not found')
     return()
   }
+
   run_gcov_one <- function(src) {
     system_check(gcov_path,
       args = c(gcov_args, src, "-p", "-o", dirname(src)),
       quiet = quiet, echo = !quiet)
-    gcov_outputs <- list.files(path, pattern = rex::rex(".gcov", end), recursive = TRUE, full.names = TRUE)
+    gcov_outputs <- list.files(".", pattern = rex::rex(".gcov", end), recursive = TRUE, full.names = TRUE)
+
+    if (!quiet) {
+      writeLines(paste0("gcov output for ", src, ":"))
+      writeLines(gcov_outputs)
+    }
+
     if (clean) {
       on.exit(unlink(gcov_outputs))
+    } else {
+      gcov_output_base <- file.path("..", "covr", src)
+      gcov_output_targets <- sub(".", gcov_output_base, gcov_outputs)
+
+      if (!quiet) {
+        writeLines(paste0("gcov output targets for ", src, ":"))
+        writeLines(gcov_output_targets)
+      }
+
+      lapply(
+        unique(dirname(gcov_output_targets)),
+        function(.x) dir.create(.x, recursive = TRUE, showWarnings = FALSE)
+      )
+
+      on.exit({
+        writeLines(c("Moving gcov outputs to covr directory.", ""))
```

Comment:
This is I guess missing a quiet conditional. Also we should probably use the equivalent cli function for the debugging output statements rather than writeLines, we already have a cli dependency anyway.